src/_pyinstaller_hooks_contrib/hooks/stdhooks/hook-pylsl.py

```diff
@@ -1,72 +1,29 @@
-there as well.
-
-See also
--  https://labstreaminglayer.readthedocs.io
--  https://github.com/labstreaminglayer/pylsl
--  https://github.com/sccn/liblsl
-"""
-
+# ------------------------------------------------------------------
+# Copyright (c) 2023 PyInstaller Development Team.
+#
+# This file is distributed under the terms of the GNU General Public
+# License (version 2.0 or later).
+#
+# The full license is available in LICENSE.GPL.txt, distributed with
+# this software.
+#
+# SPDX-License-Identifier: GPL-2.0-or-later
+# ------------------------------------------------------------------
 
-def find_liblsl_library():
-    """finds the binary lsl library.
-
-    Search order is to first try to use the path stored in the environment
-    variable PYLSL_LIB (if available), then search through the package
-    directory, and finally search the whole system.
-    """
-
-    # first try the environment variable PYLSL_LIB
-    if "PYLSL_LIB" in os.environ:
-        libfile = os.environ["PYLSL_LIB"]
-        if libfile and os.path.isfile(libfile):
-            return libfile
-
-    # this will only run if PYLSL_LIB did not resolve the library
-    os_name = platform.system()
-    if os_name in ["Windows", "Microsoft"]:
-        libsuffix = ".dll"
-    elif os_name == "Darwin":
-        libsuffix = ".dylib"
-    elif os_name == "Linux":
-        libsuffix = ".so"
-    else:
-        raise RuntimeError("unrecognized operating system:", os_name)
+import os
+from PyInstaller.utils.hooks import logger
 
-    for scope in ["package", "system"]:
-        for libprefix in ["", "lib"]:
-            for debugsuffix in ["", "-debug"]:
-                for bitness in ["", str(8 * struct.calcsize("P"))]:
-                    if scope == "package":
-                        # try in the package directory at pylsl/lib
-                        module_dir = pathlib.Path(get_module_file_attribute('pylsl')).parent
-                        libfile = os.path.join(module_dir, libprefix + "lsl" + bitness + debugsuffix + libsuffix)
-                        if libfile and os.path.isfile(libfile):
-                            return libfile
-                    elif (scope == "system") and os_name not in ["Windows", "Microsoft"]:
-                        # try on the system library path, this includes the conda library path
-                        libfile = util.find_library(libprefix + "lsl" + bitness + debugsuffix)
-                        if libfile and os.path.isfile(libfile):
-                            return libfile
-                    elif (scope == "system") and os_name == "Darwin":
-                        # try on the homebrew library path
-                        libfile = util.find_library('/opt/homebrew/lib/' + libprefix + "lsl" + bitness + debugsuffix)
-                        if libfile and os.path.isfile(libfile):
-                            return libfile
+try:
+    # the import will fail it the library cannot be found
+    from pylsl import pylsl
```

Comment:
Sorry, I should have said: Whenever a hook needs to load a 3rd party library it needs to be done in an isolated subprocess. See [here](https://pyinstaller.org/en/v5.10.1/hooks.html#module-PyInstaller.isolated). Basically put this in a function then use [isolated.call()](https://pyinstaller.org/en/v5.10.1/hooks.html#PyInstaller.isolated.call) on it.

Reply:
I have now implemented this.